lib/ride/services/audio.dart

```diff
@@ -0,0 +1,820 @@
+  Future<void> _processRideUpdates() async {
+    if (ride?.navigationIsActive != true) return;
+    if (ride?.route == null) return;
+    // If the current route is null, we won't see a rerouting but the first route.
+    if (currentRoute == null) {
+      currentRoute = ride!.route;
+      return;
+    }
+    // Rerouting
+    if (currentRoute != ride!.route && ride?.route != null) {
+      currentRoute = ride?.route;
+      if (ftts == null) await _initializeTTS();
+      ftts?.speak("Neue Route berechnet");
+    }
+
+    if (ride!.userSelectedSG != null) return;
+    if (ride!.calcCurrentSG == null) return;
+
+    // Check if the prediction is not valid anymore.
+    // If same signal group.
+    if (lastSignalGroupId == ride!.calcCurrentSGIndex?.toInt()) {
+      // Check if the current prediction is still valid.
+      // If the prediction quality is not good enough and the last prediction was good enough, inform the user.
+      if (lastPrediction?.predictionQuality != null &&
+          lastPrediction!.predictionQuality! > predictionQualityThreshold &&
+          (ride!.predictionProvider?.prediction?.predictionQuality == null ||
+              ride!.predictionProvider!.prediction!.predictionQuality! < predictionQualityThreshold) &&
+          currentSpeedAdvisoryInstructionState > 0) {
+        // Inform the user that the prediction is not valid any more.
+        _playPredictionNotValidAnymore();
+      }
+    }
+
+    lastPrediction = ride!.predictionProvider?.prediction;
+  }
+
+  /// Check if the audio instructions setting has changed.
+  Future<void> _processSettingsUpdates() async {
+    if (initialized && !settings!.audioSpeedAdvisoryInstructionsEnabled) {
+      initialized = false;
+      cleanUp();
+    } else if (!initialized && settings!.audioSpeedAdvisoryInstructionsEnabled) {
+      initialized = true;
+      _init();
+    }
+  }
+
+  /// Reset the text-to-speach instance.
+  Future<void> resetFTTS() async {
+    await ftts?.pause();
+    await ftts?.stop();
+    await ftts?.clearVoice();
+    ftts = null;
+  }
+
+  /// Process positioning updates to play audio instructions.
+  Future<void> _processPositioningUpdates() async {
+    if (settings?.audioSpeedAdvisoryInstructionsEnabled != true && settings?.audioRoutingInstructionsEnabled != true) {
+      return;
+    }
+
+    if (ride?.navigationIsActive != true) {
+      return;
+    }
+
+    if (ftts == null) {
+      await _initializeTTS();
+    }
+
+    _addSpeedValueToLastSpeedValues();
+
+    if (settings?.audioRoutingInstructionsEnabled == true) {
+      // Create navigation and speed advisory instructions.
+      await _playAudioRoutingInstruction();
+    } else if (settings?.audioSpeedAdvisoryInstructionsEnabled == true) {
+      // Create only speed advisory instructions.
+      await _playSpeedAdvisoryInstruction();
+    }
+
+    // In both modes we check if the user is waiting for green and play a countdown.
+    _checkPlayCountdownWhenWaitingForGreen();
```

Comment:
If we remove the routing feature we might also be able to add this to `_playSpeedAdvisoryInstruction`?